packages/ui/src/hooks/use-preview.ts

```diff
@@ -1,48 +1,80 @@
 import { Language } from '@logto/phrases';
 import { AppearanceMode } from '@logto/schemas';
-import { useMemo } from 'react';
+import i18next from 'i18next';
+import { useEffect, useMemo, useState } from 'react';
 
+import { Context } from '@/hooks/use-page-context';
+import initI18n from '@/i18n/init';
 import { SignInExperienceSettingsResponse } from '@/types';
 import { parseQueryParameters } from '@/utils';
+import { parseSignInExperienceSettings } from '@/utils/sign-in-experience';
 
 type PreviewConfig = {
   signInExperience: SignInExperienceSettingsResponse;
   language: Language;
-  mode: AppearanceMode;
+  mode: AppearanceMode.LightMode | AppearanceMode.DarkMode;
   platform: 'web' | 'mobile';
 };
 
-const usePreview = (): [boolean, PreviewConfig?] => {
-  const { preview, config } = parseQueryParameters(window.location.search);
+const usePreview = (context: Context): [boolean, PreviewConfig?] => {
+  const [previewConfig, setPreviewConfig] = useState<PreviewConfig>();
+  const { setTheme, setExperienceSettings, setPlatform } = context;
 
-  const previewConfig = useMemo(() => {
-    if (!preview || !config) {
+  const { preview } = parseQueryParameters(window.location.search);
+  const isPreview = useMemo(() => preview === 'true', [preview]);
```

Comment:
The `useMemo` here is not necessary, as this is not an "expensive calculation"

Reply:
good point. Seems like a redundant code I left.